x-pack/plugins/security/public/management/users/users_management_app.tsx

```diff
@@ -66,10 +66,14 @@ export const usersManagementApp = Object.freeze({
         const EditUserPageWithBreadcrumbs = () => {
           const { username } = useParams<{ username?: string }>();
 
+          // Additional decoding is a workaround for a bug in react-router's version of the `history` module.
+          // See https://github.com/elastic/kibana/issues/82440
+          const decodedUsername = username ? decodeURIComponent(username) : undefined;
```

Comment:
__issue:__ hmm, it seems while we fixed issue for usernames with `@`, we introduced it for the ones with `%`. For example for `user%not`, `decodeURIComponent` will throw `malformed URI sequence` :/ Should we temporarily create `tryDecodeURIComponent` function with a `try/catch` block inside?

Reply:
Ah good catch. I had tested with valid encoded entities (e.g. `%20`), but didn't think about the malformed sequences. Yeah I think a `tryDecodeURIComponent` makes the most sense here, at least for the time being. I'll make this change!